pyblish/lib.py

```diff
@@ -349,3 +351,27 @@ def _resolve_name(name, package, level):
     __import__(name)
 
     return sys.modules[name]
+
+
+def emit(signal, **kwargs):
+    """Searches through the registered callbacks and executes any callback that
+    matches the signal.
+    The keyword arguments (kwargs) are passed directly through from the emitter
+    to the callbacks.
+
+        Arguments:
+            signal (string): Name of signal emitted
+
+        Example:
+            def myCallback(data=None):
+                print data
+
+            pyblish.api.register_callback("someSignal", myCallback)
+            emit("someSignal", data={"something": "cool"})
+    """
+    for callback in pyblish._registered_callbacks.get(signal, []):
+        try:
+            callback(**kwargs)
+        except Exception as e:
+            traceback.print_exc(e)
+            raise Exception(e)
```

Comment:
Ah, I see. One trick is to throw an exception in the test while building it, like `assert False`, just after it prints. That way, it will throw your exception, and show you what got printed before it. Once you've seen that it prints the right things, you can remove the temp exception.